// enzyme-matchers/lib/assertions/toContainMatchingElements.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _html = require('../utils/html');

var _html2 = _interopRequireDefault(_html);

var _displayName = require('../utils/displayName');

var _displayName2 = _interopRequireDefault(_displayName);

var _name = require('../utils/name');

var _name2 = _interopRequireDefault(_name);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree. *
 *
 * @providesModule toContainMatchingElements
 * 
 */

function toContainMatchingElements(enzymeWrapper, n, selector) {
  var matches = enzymeWrapper.find(selector);
  var pass = matches.length === n;
  var nodeName = (0, _name2.default)(enzymeWrapper);

  return {
    pass: pass,
    message: 'Expected <' + nodeName + '> to contain ' + n + ' element' + (n === 1 ? '' : 's') + ' matching ' + ('"' + (0, _displayName2.default)(selector) + '" but ' + matches.length + ' ' + (matches.length === 1 ? 'was' : 'were') + ' found.'),
    negatedMessage: 'Expected <' + nodeName + '> to not contain ' + n + ' element' + (n === 1 ? '' : 's') + ' matching "' + (0, _displayName2.default)(selector) + '" but it did.',
    contextualInformation: {
      actual: 'HTML Output of <' + nodeName + '>:\n ' + (0, _html2.default)(enzymeWrapper)
    }
  };
}

exports.default = toContainMatchingElements;
module.exports = exports['default'];
